File: LeetCode/python3/260.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        x = 0
        for m in nums:
            x ^= m
        s = 1
        while (s & x) == 0:
            s <<= 1
        ret = [0, 0]
        for m in nums:
            if (m & s) == 0:
                ret[0] ^= m
            else:
                ret[1] ^= m
        return ret
    

File: LeetCode/cpp/260.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int> &nums) {
        int x = 0, s = 1;
        for (auto m:nums)
            x ^= m;
        while ((x & s) == 0)
            s <<= 1;
        vector<int> ret(2, 0);
        for (auto m:nums)
            (m & s) == 0 ? ret[0] ^= m : ret[1] ^= m;
        return ret;
    }
};
